--- target-org/webkit/Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp	2021-03-09 16:43:40.000000000 +0800
+++ target/webkit/Source/JavaScriptCore/runtime/IntlDateTimeFormat.cpp	2021-03-09 17:03:32.000000000 +0800
@@ -36,6 +36,7 @@
 #include <unicode/uenum.h>
 #include <wtf/text/StringBuilder.h>
 #include <wtf/unicode/icu/ICUHelpers.h>
+#include <sys/system_properties.h>

 namespace JSC {

@@ -100,8 +101,29 @@
 {
     String canonical;

+    UErrorCode status = U_UNDEFINED_VARIABLE;
     Vector<UChar, 32> buffer;
-    auto status = callBufferProducingFunction(ucal_getDefaultTimeZone, buffer);
+    size_t bufferLength = 32;
+
+    // [0] Try to get timezone from Android system property
+    char systemPropBuffer[2 * (PROP_VALUE_MAX + 1)] = {0};
+    if (__system_property_get("persist.sys.timezone", systemPropBuffer) != 0) {
+        size_t systemPropLength = strlen(systemPropBuffer);
+        bufferLength = systemPropLength;
+        if (bufferLength > buffer.capacity()) {
+            buffer.grow(bufferLength);
+        }
+
+        if (u_uastrncpy(buffer.data(), systemPropBuffer, buffer.capacity())) {
+            status = U_ZERO_ERROR;
+        }
+    }
+
+    // [1] Fallback to get timezone from ICU default
+    if (U_FAILURE(status)) {
+        status = callBufferProducingFunction(ucal_getDefaultTimeZone, buffer);
+    }
+    // [2] Get canonical timezone ID
     if (U_SUCCESS(status)) {
         Vector<UChar, 32> canonicalBuffer;
         status = callBufferProducingFunction(ucal_getCanonicalTimeZoneID, buffer.data(), buffer.size(), canonicalBuffer, nullptr);
