diff -aru target-org/webkit/CMakeLists.txt target/webkit/CMakeLists.txt
--- target-org/webkit/CMakeLists.txt	2021-03-09 16:46:04.000000000 +0800
+++ target/webkit/CMakeLists.txt	2021-03-09 16:48:26.000000000 +0800
@@ -191,10 +191,6 @@
     add_subdirectory(Tools)
 endif ()

-if (DEVELOPER_MODE)
-    add_subdirectory(PerformanceTests)
-endif ()
-
 # -----------------------------------------------------------------------------
 # Print the features list last, for maximum visibility.
 # -----------------------------------------------------------------------------
diff -aru target-org/webkit/Source/JavaScriptCore/CMakeLists.txt target/webkit/Source/JavaScriptCore/CMakeLists.txt
--- target-org/webkit/Source/JavaScriptCore/CMakeLists.txt	2021-03-09 16:47:58.000000000 +0800
+++ target/webkit/Source/JavaScriptCore/CMakeLists.txt	2021-03-09 16:49:53.000000000 +0800
@@ -1368,7 +1368,7 @@
 )
 target_include_directories(LLIntOffsetsExtractor SYSTEM PRIVATE ${JavaScriptCore_SYSTEM_INCLUDE_DIRECTORIES})

-add_subdirectory(shell)
+# add_subdirectory(shell)

 target_include_directories(LowLevelInterpreterLib
     PRIVATE "$<TARGET_PROPERTY:JavaScriptCore,INCLUDE_DIRECTORIES>")
diff -aru target-org/webkit/Source/cmake/OptionsJSCOnly.cmake target/webkit/Source/cmake/OptionsJSCOnly.cmake
--- target-org/webkit/Source/cmake/OptionsJSCOnly.cmake	2021-03-09 16:47:09.000000000 +0800
+++ target/webkit/Source/cmake/OptionsJSCOnly.cmake	2021-03-09 16:48:08.000000000 +0800
@@ -37,11 +37,7 @@
 set(ENABLE_WEBKIT OFF)
 set(ENABLE_WEBINSPECTORUI OFF)

-if (WIN32)
-    set(ENABLE_API_TESTS OFF)
-else ()
-    set(ENABLE_API_TESTS ON)
-endif ()
+set(ENABLE_API_TESTS OFF)

 if (WTF_CPU_ARM OR WTF_CPU_MIPS)
     SET_AND_EXPOSE_TO_BUILD(USE_CAPSTONE TRUE)
