--- target-org/webkit/Source/JavaScriptCore/runtime/DatePrototype.cpp	2019-11-01 00:14:03.000000000 +0800
+++ target/webkit/Source/JavaScriptCore/runtime/DatePrototype.cpp	2020-07-20 11:40:00.000000000 +0800
@@ -246,7 +246,7 @@
     int year = gdt.year();
     bool yearNeedsOffset = year < 1900 || year > 2038;
     if (yearNeedsOffset)
-        localTM.tm_year = equivalentYearForDST(year) - 1900;
+        localTM.tm_year = WTF::equivalentYearForDST(year) - 1900;

 #if HAVE(LANGINFO_H)
     // We do not allow strftime to generate dates with 2-digits years,
